WPMedia/engine-theme-sdk
# [RC] TMEDIA-98 - Meta Value 'title' should override page title logic

## Description

Page Metadata `title` value should override any other pre defined title logic 

## Jira Ticket
- [TMEDIA-98](https://arcpublishing.atlassian.net/browse/PEN-1822)

## Acceptance Criteria

1. If a value is entered for title in the PageBuilder Editor metadata (i.e., an “override title”) then that value should be used for the <title>, rather than any automatically generated title 
2. If a value is entered for og:title in the PageBuilder Editor metadata (i.e., an “override og title”) then that value should be used for the og:title, rather than any automatically generated og title 
3. If a value is entered for twitter:title in the PageBuilder Editor metadata (i.e., an “override twitter title”) then that value should be used for the twitter:title, rather than any automatically generated twitter title 
4. Note that each override only affects the corresponding single metadata tag – for example, if the only override on a page is title, that should only should affect the <title>, and og:title and twitter:title should continue to be generated automatically
5. Tests and documentation are updated as necessary for this new functionality

## Test Steps

1. Checkout this branch `git checkout TMEDIA-98-use-page-title-override`
2. Ensure fusion repo has `ENGINE_SDK_REPO` set to the correct path for your local copy of this repo
3. Verify each of the following page use the `title` page metadata value
    * http://localhost/local/?_website=the-gazette
    * http://localhost/video/2019/12/12/castle-church-tower/?_website=the-gazette
    * http://localhost/gallery/2019/12/12/winter-snow-scenes-gallery/?_website=the-gazette
    * http://localhost/author/sara-carothers/?_website=the-gazette
    * http://localhost/tags/beaches/?_website=the-gazette
    * http://localhost/about-us/?_website=the-gazette

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
